server/lib/host-metrics.js

```diff
@@ -222,38 +222,37 @@ ORDER BY DATE_TRUNC(:timeUnit, t1."createdAt")`,
     },
   );
 
-  let legacyResults = [];
-  const newHostFeeIntroductionDate = new Date('2021-01-01T00:00:00.000Z');
-  if (startDate < newHostFeeIntroductionDate) {
-    legacyResults = await sequelize.query(
-      `SELECT SUM(t1."hostFeeInHostCurrency") as "_amount", t1."hostCurrency" as "_currency", DATE_TRUNC(:timeUnit, t1."createdAt") as "date"
+  const legacyResults = await sequelize.query(
+    `SELECT SUM(t1."hostFeeInHostCurrency") as "_amount", t1."hostCurrency" as "_currency", DATE_TRUNC(:timeUnit, t1."createdAt") as "date"
 FROM "Transactions" as t1
 WHERE t1."HostCollectiveId" = :HostCollectiveId
 AND t1."createdAt" >= :startDate AND t1."createdAt" <= :endDate
 AND NOT (t1."type" = 'DEBIT' AND t1."kind" = 'ADDED_FUNDS')
 AND t1."deletedAt" IS NULL
 GROUP BY t1."hostCurrency", DATE_TRUNC(:timeUnit, t1."createdAt")
 ORDER BY DATE_TRUNC(:timeUnit, t1."createdAt")`,
-      {
-        replacements: { HostCollectiveId: host.id, ...computeDates(startDate, endDate), timeUnit },
-        type: sequelize.QueryTypes.SELECT,
-      },
-    );
-  }
+    {
+      replacements: { HostCollectiveId: host.id, ...computeDates(startDate, endDate), timeUnit },
+      type: sequelize.QueryTypes.SELECT,
+    },
+  );
 
   const newTimeSeries = await convertCurrencyForTimeSeries(newResults, host.currency);
   const legacyTimeSeries = await convertCurrencyForTimeSeries(legacyResults, host.currency);
+
   const mergedTimeSeries = [...newTimeSeries.map(point => ({ ...point, amount: Math.abs(point.amount) }))];
 
   // Merge legacy time series with new time series
-  legacyTimeSeries.forEach(point => {
-    const existingDataPoint = mergedTimeSeries.find(({ date }) => point.date === date);
+  for (const point of legacyTimeSeries) {
```

Comment:
When troubleshooting I was more confident of the behavior with `for` than with `forEach`.

Reply:
I prefer the non-async behavior but that should not change anything there.